library/ssl_cli.c

```diff
@@ -4361,7 +4361,18 @@ int mbedtls_ssl_handshake_client_step( mbedtls_ssl_context *ssl )
            ret = mbedtls_ssl_parse_delayed_certificate_verify( ssl, authmode,
                                    ssl->session_negotiate->peer_cert, rs_ctx );
            if( ret != 0 )
+           {
                break;
+           }
+           if( ret != 0 )
+           {
+               break;
+           }
+           else
```

Comment:
It seems you will always report an error

Reply:
You're right. Fixing